stripe/src/test/java/com/stripe/android/StripeTest.java

```diff
@@ -248,6 +253,204 @@ public void createBankAccountTokenSynchronous_withValidBankAccount_returnsToken(
+            assertNotNull(giropaySource.getId());
+            assertEquals(Source.GIROPAY, giropaySource.getType());
+            assertNotNull(giropaySource.getSourceTypeData());
+            assertNotNull(giropaySource.getOwner());
+            assertNotNull(giropaySource.getRedirect());
+            assertEquals("Mr. X", giropaySource.getOwner().getName());
+            assertEquals("example://redirect", giropaySource.getRedirect().getReturnUrl());
+        } catch (StripeException stripeEx) {
+            fail("Unexpected error: " + stripeEx.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void createSourceSynchronous_withSepaDebitParams_passesIntegrationTest() {
+        Stripe stripe = new Stripe(mContext);
+        String validIban = "DE89370400440532013000";
+        SourceParams params = SourceParams.createSepaDebitParams(
+                "Sepa Account Holder",
+                validIban,
+                "123 Main St",
+                "Eureka",
+                "90210",
+                "EI");
+        try {
+            Source sepaDebitSource =
+                    stripe.createSourceSynchronous(params, FUNCTIONAL_SOURCE_PUBLISHABLE_KEY);
+            assertNotNull(sepaDebitSource);
+            assertNotNull(sepaDebitSource.getClientSecret());
+            assertNotNull(sepaDebitSource.getId());
+            assertEquals(Source.SEPA_DEBIT, sepaDebitSource.getType());
+            assertNotNull(sepaDebitSource.getSourceTypeData());
+            assertNotNull(sepaDebitSource.getOwner());
+            assertNotNull(sepaDebitSource.getOwner().getAddress());
+            assertEquals("Eureka", sepaDebitSource.getOwner().getAddress().getCity());
+            assertEquals("90210", sepaDebitSource.getOwner().getAddress().getPostalCode());
+            assertEquals("123 Main St", sepaDebitSource.getOwner().getAddress().getLine1());
+            assertEquals("EI", sepaDebitSource.getOwner().getAddress().getCountry());
+            assertEquals("Sepa Account Holder", sepaDebitSource.getOwner().getName());
+        } catch (StripeException stripeEx) {
+            fail("Unexpected error: " + stripeEx.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void createSourceSynchronous_withIDealParams_passesIntegrationTest() {
+        Stripe stripe = new Stripe(mContext);
+        SourceParams params = SourceParams.createIdealParams(
+                5500L,
+                "Bond",
+                "example://return",
+                "A statement description",
+                "rabobank");
+        try {
+            Source idealSource =
+                    stripe.createSourceSynchronous(params, FUNCTIONAL_SOURCE_PUBLISHABLE_KEY);
+            assertNotNull(idealSource);
+            assertNotNull(idealSource.getClientSecret());
+            assertNotNull(idealSource.getId());
+            assertEquals(Source.IDEAL, idealSource.getType());
+            assertNotNull(idealSource.getSourceTypeData());
```

Comment:
I'm avoiding checking the type data for now -- only using rabobank because we have a fixed enum set of values that are allowed to be passed in that.